xray/commands/audit/generic/auditmanager.go

```diff
@@ -88,9 +100,75 @@ func (params *Params) SetMinSeverityFilter(minSeverityFilter string) *Params {
 	return params
 }
 
-// GenericAudit audits all the projects found in the given workingDirs
-func GenericAudit(params *Params) (results []services.ScanResponse, isMultipleRoot bool, err error) {
-	if err = coreutils.ValidateMinimumVersion(coreutils.Xray, params.xrayVersion, utils.GraphScanMinXrayVersion); err != nil {
+func (params *Params) SetXrayVersion(version string) *Params {
+	params.xrayVersion = version
+	return params
+}
+
+// Runs an audit scan based on the provided auditParams.
+// Returns an audit Results object containing all the scan results.
+// If the current server is entitled for JAS, the advanced security results will be included in the scan results.
+func RunAudit(auditParams *Params) (results *Results, err error) {
+	serverDetails, err := auditParams.ServerDetails()
+	if err != nil {
+		return
+	}
+	isEntitled, xrayVersion, err := isEntitledForJas(serverDetails)
+	if err != nil {
+		return
+	}
+	auditParams.SetXrayVersion(xrayVersion)
+
+	errGroup := new(errgroup.Group)
+	if isEntitled {
+		// Download (if needed) the analyzer manager in a background routine.
+		errGroup.Go(rtutils.DownloadAnalyzerManagerIfNeeded)
+	}
+
+	// The audit scan doesn't require the analyzer manager, so it can run separately from the analyzer manager download routine.
+	scanResults, isMultipleRootProject, auditError := genericAudit(auditParams)
+
+	// Wait for the Download of the AnalyzerManager to complete.
+	if err = errGroup.Wait(); err != nil {
+		return
+	}
+
+	extendedScanResults := &clientUtils.ExtendedScanResults{XrayResults: scanResults}
+	// Try to run contextual analysis only if the user is entitled for advance security
+	if isEntitled {
+		extendedScanResults, err = jas.GetExtendedScanResults(scanResults, auditParams.FullDependenciesTree(), serverDetails)
+		if err != nil {
+			return
+		}
+	}
+	results = &Results{
+		IsMultipleRootProject: isMultipleRootProject,
+		AuditError:            auditError,
+		ExtendedScanResults:   extendedScanResults,
+	}
+	return
+}
+
+func isEntitledForJas(serverDetails *config.ServerDetails) (entitled bool, xrayVersion string, err error) {
+	xrayManager, xrayVersion, err := commandsutils.CreateXrayServiceManagerAndGetVersion(serverDetails)
+	if err != nil {
+		return
+	}
+	err = coreutils.ValidateMinimumVersion(coreutils.Xray, xrayVersion, clientUtils.EntitlementsMinVersion)
```

Comment:
Modifying the `ValidateMinimumVersion` to return bool instead of err should simply the usage a lot.

Reply:
done